primefaces/src/main/java/org/primefaces/component/colorpicker/ColorPickerRenderer.java

```diff
@@ -28,32 +28,39 @@
 import java.util.regex.Pattern;
 
 import javax.faces.component.UIComponent;
+import javax.faces.component.UINamingContainer;
 import javax.faces.context.FacesContext;
 import javax.faces.context.ResponseWriter;
 import javax.faces.convert.Converter;
 
 import org.primefaces.renderkit.InputRenderer;
-import org.primefaces.util.ComponentUtils;
-import org.primefaces.util.Constants;
-import org.primefaces.util.HTML;
-import org.primefaces.util.WidgetBuilder;
+import org.primefaces.util.*;
 
 public class ColorPickerRenderer extends InputRenderer {
 
-    private static final Pattern COLOR_HEX_PATTERN = Pattern.compile("([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})");
+    private static final Pattern COLOR_PATTERN = Pattern.compile(
```

Comment:
I'm not sure if this regex is the best thing for readability and maintainability. I would have created a validation method based on the start of the value. Anyway, this will probably never change (will it?).

Reply:
The old code used a regex to defend invalid values and I don't think checking RGB etc is enough as rgbXXX would be considered a valid value.  So I kept it a regex check and it works for all the color types!